modules/installation-osp-local-disk-deployment.adoc

```diff
@@ -0,0 +1,355 @@
+for machine in ${cp_machines}; do
+  if ! oc get machine -n openshift-machine-api "${machine}" -o jsonpath='{.spec.providerSpec.value.additionalBlockDevices}' | grep -q 'etcd'; then
+	exit 1
+  fi <3>
+done
+----
+<1> Retrieves the machines running in the cluster.
+<2> Checks that there are 3 machines running.
+<3> Checks that 3 `etcd` devices are running on the 3 machines.
+
+. Create a file named `98-var-lib-etcd.yaml` by using the following content:
+
++
+[%collapsible]
+====
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 98-var-lib-etcd
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - contents: |
+          [Unit]
+          Description=Make File System on /dev/vdb
+          DefaultDependencies=no
+          BindsTo=dev-vdb.device
+          After=dev-vdb.device var.mount
+          Before=systemd-fsck@dev-vdb.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecStart=/usr/sbin/mkfs.xfs -f /dev/vdb
+          TimeoutSec=0
+
+          [Install]
+          WantedBy=var-lib-containers.mount
+        enabled: true
+        name: systemd-mkfs@dev-vdb.service
+      - contents: |
+          [Unit]
+          Description=Mount /dev/vdb to /var/lib/etcd
+          Before=local-fs.target
+          Requires=systemd-mkfs@dev-vdb.service
+          After=systemd-mkfs@dev-vdb.service var.mount
+
+          [Mount]
+          What=/dev/vdb
+          Where=/var/lib/etcd
+          Type=xfs
+          Options=defaults,prjquota
+
+          [Install]
+          WantedBy=local-fs.target
+        enabled: true
+        name: var-lib-etcd.mount
+      - contents: |
+          [Unit]
+          Description=Sync etcd data if new mount is empty
+          DefaultDependencies=no
+          After=var-lib-etcd.mount var.mount
+          Before=crio.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecCondition=/usr/bin/test ! -d /var/lib/etcd/member
+          ExecStart=/usr/sbin/setenforce 0
+          ExecStart=/bin/rsync -ar /sysroot/ostree/deploy/rhcos/var/lib/etcd/ /var/lib/etcd/
+          ExecStart=/usr/sbin/setenforce 1
+          TimeoutSec=0
+
+          [Install]
+          WantedBy=multi-user.target graphical.target
+        enabled: true
+        name: sync-var-lib-etcd-to-etcd.service
+      - contents: |
+          [Unit]
+          Description=Restore recursive SELinux security contexts
+          DefaultDependencies=no
+          After=var-lib-etcd.mount
+          Before=crio.service
+
+          [Service]
+          Type=oneshot
+          RemainAfterExit=yes
+          ExecStart=/sbin/restorecon -R /var/lib/etcd/
+          TimeoutSec=0
+
+          [Install]
+          WantedBy=multi-user.target graphical.target
+        enabled: true
+        name: restorecon-var-lib-etcd.service
```

Comment:
This will run unconditionally on every boot, but it only needs to happen when we do the original content restoration. We could move this into the sync etcd service above.

Reply:
Can you explain more by commenting where above you'd like this included?

Reply:
I think we need to do some engineering work here to recommend a working configuraiton.